src/context/typecore.ml

```diff
@@ -222,7 +222,31 @@ let add_local ctx k n t p =
 	ctx.locals <- PMap.add n v ctx.locals;
 	v
 
+let check_identifier_name ctx name kind p =
+	if starts_with name '$' then
+		display_error ctx (kind ^ " names starting with a dollar are not allowed") p
```

Comment:
Hmm, technically might have to uppercase the first letter of `kind` here to make it consistent with other errors (which usually start uppercase).

Reply:
Agreed